pkg/controller/machine/machine_controller.go

```diff
@@ -458,6 +465,9 @@ func (r *Reconciler) ensureMachineHasNodeReadyCondition(machine *clusterv1alpha1
 			return nil
 		}
 	}
+
+	r.metrics.Provisioning.Observe(time.Until(machine.CreationTimestamp.Time).Abs().Seconds())
```

Comment:
as I understood in call yesterday, the `reconcile` function (which calls `ensureMachineHasNodeReadyCondition`) is called multiple times and can even be called after we already observed the provisioning time for a given Machine - will this lead to multiple observations per Machine?

Reply:
`ensureMachineHasNodeReadyCondition` gets called multiple times, but the if condition just above the `.Observe` prevents multiple observations:

```
		if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
			return nil
		}
```

Reply:
this is not combined with the Node status, correct? So if the Node related to the Machine goes offline and later comes back online, going `NodeReady`, this does not affect this check?

Reply:
The `NodeReady` condition is set on the Machine. E.g. rebooting the VM may cause a `NotReady` status of the node, but does not affect this check